extensions/parquet/compression/src/main/java/io/deephaven/parquet/compress/DeephavenCompressorAdapterFactory.java

```diff
@@ -45,73 +41,109 @@ public class DeephavenCodecFactory {
                 }
             }).collect(Collectors.toList());
 
-    private static volatile DeephavenCodecFactory INSTANCE;
+    private static volatile DeephavenCompressorAdapterFactory INSTANCE;
 
-    public static synchronized void setInstance(DeephavenCodecFactory factory) {
+    public static synchronized void setInstance(DeephavenCompressorAdapterFactory factory) {
         if (INSTANCE != null) {
             throw new IllegalStateException("Can't assign an instance when one is already set");
         }
         INSTANCE = factory;
     }
 
-    public static DeephavenCodecFactory getInstance() {
+    public static DeephavenCompressorAdapterFactory getInstance() {
         if (INSTANCE == null) {
-            synchronized (DeephavenCodecFactory.class) {
+            synchronized (DeephavenCompressorAdapterFactory.class) {
                 if (INSTANCE == null) {
-                    INSTANCE = new DeephavenCodecFactory(CODECS);
+                    INSTANCE = new DeephavenCompressorAdapterFactory(CODECS);
                 }
             }
         }
         return INSTANCE;
     }
 
-    public static class CodecWrappingCompressor implements Compressor {
+    public static class CodecWrappingCompressorAdapter implements CompressorAdapter {
         private final CompressionCodec compressionCodec;
 
-        private CodecWrappingCompressor(CompressionCodec compressionCodec) {
+        private boolean innerCompressorPooled;
+        private Compressor innerCompressor;
+
+        public boolean innerDecompressorPooled;
+        private Decompressor innerDecompressor;
+
+        private CodecWrappingCompressorAdapter(CompressionCodec compressionCodec) {
             this.compressionCodec = compressionCodec;
+            this.innerCompressor = compressionCodec.createCompressor();
```

Comment:
I dont think this line makes any sense. Either a) it returns non-null, and so compress(OutputStream) below can never borrow from the pool, or b) it returns null, and CodecPoo.getCompressor will also return null, as will the same call to compressionCodec.createCompressor().

With this line removed, ParquetTableReadWriteTest will correctly stop in close()'s first if block to return the codec to the pool, otherwise it never stops there.